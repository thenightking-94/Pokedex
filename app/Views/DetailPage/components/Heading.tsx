import { Grid, useMediaQuery } from '@mui/material';
import { useRouter } from 'next/router';
import { ReactElement } from 'react';
import { Bars, ThreeCircles } from 'react-loader-spinner';
import { returnLinearGradientStyles, formatPokemonId } from '../../../Utils/Utils';
import BackIcon from '../../Icons-svg/backIcon';
import CloseIcon from '../../Icons-svg/closeIcon';
import ForwardIcon from '../../Icons-svg/forwardIcon';

export default function Heading ({detailData, color1, color2, name, id, flText, toggleReadMore, readMore}): ReactElement{
    const router = useRouter()
    const isMobile = useMediaQuery('(max-width:720px)')

    return(
        <>
            {isMobile ?
                <div className='mobile-heading'>
                    <div className='title-id'>
                         <div className='title'>
                            {name.toUpperCase()}
                         </div>
                         <div className='id'>
                             {formatPokemonId(id)}
                         </div>
                    </div>
                    <div onClick={()=>{ router.push('/') }} className='close-icon'><CloseIcon/></div>
                </div> : null
            }
            <Grid container>
                    <Grid item xs={6} sm={4}>
                        <div className={isMobile? 'image-card-mobile' : 'image-card'} style={returnLinearGradientStyles(color1,color2)}>
                            {!detailData && <Bars height="60" width="60" color="grey" ariaLabel="bars-loading" visible={true} />}
                            {!!detailData && <img alt={name} aria-labelledby={name} src={detailData?.sprites?.other?.dream_world?.front_default} className='det-img'/>}
                        </div>
                    </Grid>
                    <Grid item xs={6} sm={8}>
                        {!isMobile && 
                            <div className='heading-nav'>
                                <div className='pokemon-name'>{name.toUpperCase()}</div>
                                <div className='pokemon-id'>{formatPokemonId(id)}</div>
                                <div className='nav'>
                                    <div onClick={()=>{router.push(`/detail/${Number(id)-1}`)}} className='back-icon'><BackIcon/></div>
                                    <div onClick={()=>{ router.push('/') }} className='close-icon'><CloseIcon/></div>
                                    <div onClick={()=>{router.push(`/detail/${Number(id)+1}`)}} className='forward-icon'><ForwardIcon/></div>
                                </div>
                            </div>}
                        {flText? 
                                <div className={isMobile? 'text-section-mobile' :'text-section'}>
                                    {flText} {!!readMore && <span onClick={toggleReadMore} className='read-more'><u>{'  read more'}</u></span>}
                                    {!readMore && <span onClick={toggleReadMore} className='read-more'><u>{'  read less'}</u></span>}
                                 </div>
                                 :
                                 <div className='loader'>
                                    <ThreeCircles
                                        height="70"
                                        width="70"
                                        color="#4fa94d"
                                        visible={true}
                                        ariaLabel="three-circles-rotating"
                                        outerCircleColor="#2E3156"
                                        innerCircleColor="#CFB7ED"
                                        middleCircleColor="#F4D1A6"
                                        />
                                 </div>
                        }
                    </Grid>
             </Grid>
        </>
       
    )

}