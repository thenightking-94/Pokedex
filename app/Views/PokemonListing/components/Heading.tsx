import { useMediaQuery } from '@mui/material';
import { useRouter } from 'next/router';
import React from 'react';

export default function Heading (){
const router = useRouter()
const isMobile = useMediaQuery('(max-width:720px)')
    return(
        <div onClick={() => { router.push('/') }} className={isMobile? 'heading-mobile' : 'heading'}>
            <div title='Click here to go to homepage' className='label'>Pokédex</div>
            <div className='small-label'>Search for any Pokémon that exists on the planet</div>
        </div>
    )

}