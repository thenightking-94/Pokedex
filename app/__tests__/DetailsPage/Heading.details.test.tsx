import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import renderer from "react-test-renderer";
import Heading from '../../Views/DetailPage/components/Heading'

const queryClient = new QueryClient()

const props = {
    detailData : null,  color1 : '#dcdcdc', color2 : '#FCC1B0', name : 'Charizard', id : '6', flText : null, toggleReadMore: null, readMore: false}

describe('Heading', () => {
  it('renders correctly', () => {
    render(
    <QueryClientProvider client={queryClient}>
         <Heading {...props} />
    </QueryClientProvider>
   )
  })
})

describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create( 
      <QueryClientProvider client={queryClient}>
        <Heading {...props}/>
     </QueryClientProvider>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });