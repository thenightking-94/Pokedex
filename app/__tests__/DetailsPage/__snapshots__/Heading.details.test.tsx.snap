// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-1xd74lj-MuiGrid-root"
  >
    <div
      className="image-card"
      style={
        {
          "backgroundImage": "linear-gradient(to bottom, #dcdcdc, #FCC1B0)",
        }
      }
    >
      <div
        aria-busy={true}
        aria-label="bars-loading"
        data-testid="bars-loading"
        role="status"
        style={
          {
            "display": "flex",
          }
        }
      >
        <svg
          data-testid="bars-svg"
          fill="grey"
          height="60"
          viewBox="0 0 135 140"
          width="60"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            height="120"
            rx="6"
            width="15"
            y="10"
          >
            <animate
              attributeName="height"
              begin="0.5s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="120;110;100;90;80;70;60;50;40;140;120"
            />
            <animate
              attributeName="y"
              begin="0.5s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="10;15;20;25;30;35;40;45;50;0;10"
            />
          </rect>
          <rect
            height="120"
            rx="6"
            width="15"
            x="30"
            y="10"
          >
            <animate
              attributeName="height"
              begin="0.25s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="120;110;100;90;80;70;60;50;40;140;120"
            />
            <animate
              attributeName="y"
              begin="0.25s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="10;15;20;25;30;35;40;45;50;0;10"
            />
          </rect>
          <rect
            height="140"
            rx="6"
            width="15"
            x="60"
          >
            <animate
              attributeName="height"
              begin="0s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="120;110;100;90;80;70;60;50;40;140;120"
            />
            <animate
              attributeName="y"
              begin="0s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="10;15;20;25;30;35;40;45;50;0;10"
            />
          </rect>
          <rect
            height="120"
            rx="6"
            width="15"
            x="90"
            y="10"
          >
            <animate
              attributeName="height"
              begin="0.25s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="120;110;100;90;80;70;60;50;40;140;120"
            />
            <animate
              attributeName="y"
              begin="0.25s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="10;15;20;25;30;35;40;45;50;0;10"
            />
          </rect>
          <rect
            height="120"
            rx="6"
            width="15"
            x="120"
            y="10"
          >
            <animate
              attributeName="height"
              begin="0.5s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="120;110;100;90;80;70;60;50;40;140;120"
            />
            <animate
              attributeName="y"
              begin="0.5s"
              calcMode="linear"
              dur="1s"
              repeatCount="indefinite"
              values="10;15;20;25;30;35;40;45;50;0;10"
            />
          </rect>
        </svg>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-8 css-112spkq-MuiGrid-root"
  >
    <div
      className="heading-nav"
    >
      <div
        className="pokemon-name"
      >
        CHARIZARD
      </div>
      <div
        className="pokemon-id"
      >
        006
      </div>
      <div
        className="nav"
      >
        <div
          className="back-icon"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 22 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.92498 8.49641L10.2785 8.14286L9.57143 7.43575L9.21787 7.7893L9.92498 8.49641ZM6.71428 11L6.36073 10.6464L6.00718 11L6.36073 11.3536L6.71428 11ZM9.21787 14.2107L9.57143 14.5642L10.2785 13.8571L9.92498 13.5036L9.21787 14.2107ZM9.21787 7.7893L6.36073 10.6464L7.06784 11.3536L9.92498 8.49641L9.21787 7.7893ZM6.36073 11.3536L9.21787 14.2107L9.92498 13.5036L7.06784 10.6464L6.36073 11.3536ZM6.71428 11.5L16 11.5V10.5L6.71428 10.5V11.5ZM11 1.5C16.2467 1.5 20.5 5.75329 20.5 11H21.5C21.5 5.20101 16.799 0.5 11 0.5V1.5ZM1.5 11C1.5 5.75329 5.75329 1.5 11 1.5V0.5C5.20101 0.5 0.5 5.20101 0.5 11H1.5ZM11 20.5C5.75329 20.5 1.5 16.2467 1.5 11H0.5C0.5 16.799 5.20101 21.5 11 21.5V20.5ZM11 21.5C16.799 21.5 21.5 16.799 21.5 11H20.5C20.5 16.2467 16.2467 20.5 11 20.5V21.5Z"
              fill="#2E3156"
            />
          </svg>
        </div>
        <div
          className="close-icon"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="26"
            viewBox="0 0 26 26"
            width="26"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.85714 7.85714L18.1429 18.1429M7.85714 18.1429L18.1429 7.85714M13 25C6.37258 25 1 19.6274 1 13C1 6.37258 6.37258 1 13 1C19.6274 1 25 6.37258 25 13C25 19.6274 19.6274 25 13 25Z"
              stroke="#2E3156"
            />
          </svg>
        </div>
        <div
          className="forward-icon"
          onClick={[Function]}
        >
          <svg
            fill="none"
            height="24"
            viewBox="0 0 22 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.0748 13.5036L11.7212 13.8571L12.4283 14.5642L12.7819 14.2107L12.0748 13.5036ZM15.2855 11L15.639 11.3536L15.9926 11L15.639 10.6464L15.2855 11ZM12.7819 7.7893L12.4283 7.43575L11.7212 8.14286L12.0748 8.49641L12.7819 7.7893ZM12.7819 14.2107L15.639 11.3536L14.9319 10.6464L12.0748 13.5036L12.7819 14.2107ZM15.639 10.6464L12.7819 7.7893L12.0748 8.49641L14.9319 11.3536L15.639 10.6464ZM15.2855 10.5H5.99976V11.5H15.2855V10.5ZM10.9998 20.5C5.75305 20.5 1.49976 16.2467 1.49976 11H0.499756C0.499756 16.799 5.20077 21.5 10.9998 21.5V20.5ZM20.4998 11C20.4998 16.2467 16.2465 20.5 10.9998 20.5V21.5C16.7987 21.5 21.4998 16.799 21.4998 11H20.4998ZM10.9998 1.5C16.2465 1.5 20.4998 5.7533 20.4998 11H21.4998C21.4998 5.20101 16.7987 0.5 10.9998 0.5V1.5ZM10.9998 0.5C5.20077 0.5 0.499756 5.20101 0.499756 11H1.49976C1.49976 5.7533 5.75305 1.5 10.9998 1.5V0.5Z"
              fill="#2E3156"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="loader"
    >
      <div
        aria-busy={true}
        aria-label="three-circles-rotating"
        className=""
        data-testid="three-circles-wrapper"
        role="status"
        style={
          {
            "display": "flex",
          }
        }
      >
        <svg
          data-testid="three-circles-svg"
          enableBackground="new 0 0 100 100"
          height="70"
          version="1.1"
          viewBox="0 0 100 100"
          width="70"
          xmlSpace="preserve"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M31.6,3.5C5.9,13.6-6.6,42.7,3.5,68.4c10.1,25.7,39.2,38.3,64.9,28.1l-3.1-7.9c-21.3,8.4-45.4-2-53.8-23.3
  c-8.4-21.3,2-45.4,23.3-53.8L31.6,3.5z"
            fill="#2E3156"
          >
            <animateTransform
              attributeName="transform"
              attributeType="XML"
              dur="2s"
              from="0 50 50"
              repeatCount="indefinite"
              to="360 50 50"
              type="rotate"
            />
          </path>
          <path
            d="M42.3,39.6c5.7-4.3,13.9-3.1,18.1,2.7c4.3,5.7,3.1,13.9-2.7,18.1l4.1,5.5c8.8-6.5,10.6-19,4.1-27.7
  c-6.5-8.8-19-10.6-27.7-4.1L42.3,39.6z"
            fill="#F4D1A6"
          >
            <animateTransform
              attributeName="transform"
              attributeType="XML"
              dur="1s"
              from="0 50 50"
              repeatCount="indefinite"
              to="-360 50 50"
              type="rotate"
            />
          </path>
          <path
            d="M82,35.7C74.1,18,53.4,10.1,35.7,18S10.1,46.6,18,64.3l7.6-3.4c-6-13.5,0-29.3,13.5-35.3s29.3,0,35.3,13.5
  L82,35.7z"
            fill="#CFB7ED"
          >
            <animateTransform
              attributeName="transform"
              attributeType="XML"
              dur="2s"
              from="0 50 50"
              repeatCount="indefinite"
              to="360 50 50"
              type="rotate"
            />
          </path>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
